src/service/model/judge.go

```diff
@@ -4,6 +4,41 @@ import (
 	"strings"
 )
 
+// Should contains a pirority definition
+// Ex. ComileError > RuntimeError > TimeLimitExceeded > MemoryLimitExceeded > WrongAnswer > Accepted
+type JudgeVerdict string
+
+const (
+	JudgeVerdictAccepted            JudgeVerdict = "accepted"
+	JudgeVerdictWrongAnswer         JudgeVerdict = "wrong_answer"
+	JudgeVerdictTimeLimitExceeded   JudgeVerdict = "time_limit_exceeded"
+	JudgeVerdictMemoryLimitExceeded JudgeVerdict = "memory_limit_exceeded"
+	JudgeVerdictRuntimeError        JudgeVerdict = "runtime_error"
+	JudgeVerdictCompileError        JudgeVerdict = "compile_error" // Only for main verdict
+	JudgeVerdictCancelled           JudgeVerdict = "cancelled"     // Judge will be cancelled if some point results in Runtime error, Time limit exceeded, Memory limit exceeded
+)
+
+type JudgeResult struct {
+	MainVerdict    JudgeVerdict         `json:"verdict"`        // A merge of all TestPoints' verdict, according to the pirority
+	Detail         string               `json:"detail"`         // A brief description of the result
+	TestPointCount uint64               `json:"testPointCount"` // Won't be stored in database
+	TestPointMap   map[string]TestPoint `json:"testPoints"`     // Won't be stored in database
+	TestPointsJson string               `json:"-"`              // Used to store TestPoints in database
+	AverageTimeMs  uint64               `json:"averageTimeMs"`  // Won't be stored in database
+}
```

Comment:
Memory cost should also be considered, while MaxTime & MaxMemory are more usable than average ones.

Reply:
I‘m not sure if we have an accurate memory monitor in judger now.